Source/DafnyLanguageServer/Workspace/ProjectManager.cs

```diff
@@ -168,7 +165,13 @@ public class ProjectManager : IDisposable {
   }
 
   private ExecutionEngine GetBoogie() {
-    return boogieEngine ?? new ExecutionEngine(options, cache, scheduler);
+    if (options.Get(ReuseSolvers)) {
+      boogieEngine ??= new ExecutionEngine(options, cache, scheduler);
+    } else {
+      boogieEngine?.Dispose();
```

Comment:
Good call. I forgot about that.

Reply:
Failing test about process counts alerted me to it